components/ui/common/form/datePicker/index.tsx

```
@@ -9,32 +9,65 @@ import setMinutes from "date-fns/setMinutes";
 export default function DatePickerComponent({
   startDate: initialStartDate,
   endDate: initialEndDate,
   onChange,
-}) {
-  const startingDate = initialStartDate
-    ? new Date(initialStartDate)
+}: DatePickerComponentProps) {
+  const parsedStartDate = initialStartDate instanceof Date ? initialStartDate : initialStartDate ? new Date(initialStartDate) : null;
+  const parsedEndDate = initialEndDate instanceof Date ? initialEndDate : initialEndDate ? new Date(initialEndDate) : null;
+
+  const startingDate = parsedStartDate && !isNaN(parsedStartDate.getTime())
```

Biome (1.9.4)

[error] isNaN is unsafe. It attempts a type coercion. Use Number.isNaN instead.

See the MDN documentation for more details.
Unsafe fix: Use Number.isNaN instead.


(lint/suspicious/noGlobalIsNan)